src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> indexNode = getByIndex(index);
+        if (indexNode.prev == null) {
+            linkFirst(value, indexNode);
+        } else {
+            linkMiddle(value, indexNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        final T removedItem = getByIndex(index).item;
+        getByIndex(index).item = value;
+        return removedItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removedItem = getByIndex(index);
+        unlink(removedItem);
+        return removedItem.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> removedNode = findByIndex(object);
+        if (removedNode != null) {
+            unlink(removedNode);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void validateIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + " doesn't exist");
+        }
+    }
+
+    private Node<T> findByIndex(T searchingItem) {
```

Comment:
method name does not match to what it do
```suggestion
    private Node<T> findByItem(T searchingItem) {
```